### ISSUE TITLE
Support ! after type/scope to indicate a breaking change

### ISSUE THREAD

**Issue by benswinburne** [intuit/auto#1597]
**Describe the solution you'd like**

I created a pull request today which included ! in the type, for example "feat!: implemented config v2 syntax" which was seen as a patch by the conventional commit plugin (which I suppose is actually a bug as it should be a minor?)

According to the conventional commits website https://www.conventionalcommits.org/en/v1.0.0/

> BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type.

It'd be great if ! was supported to indicate a breaking change.

**Describe alternatives you've considered**

Alternative solutions would be just to not use the ! syntax- perhaps if this isn't something which is likely to be considered I will submit a PR to the readme to indicate that it's not supported and would clash with the `BREAKING CHANGE:` commit footer.

**Additional context**

```
refactor!: drop support for Node 6

BREAKING CHANGE: refactor to use JavaScript features not available in Node 6.
```

Taking the example from the conventional commits page, the above example creates a patch version despite the ! and the footer.

I had a look at the way Auto inspects commits and it appears to use `parse-commit-message` which only checks for breaking changes using the following.

```
function isBreakingChange({ header, body, footer }) {
  const re = /^BREAKING\s+CHANGES?:\s+/;
  return (
    re.test(header.subject) || re.test(body || '') || re.test(footer || '')
  );
}
```

The repo for that library is now read only so I'm not sure whether or not support for the ! could be added there first or if it'd need to be forked swapped out perhaps?

**Comment by hipstersmoothie** [comment]
Do you have any suggestions for a replacement dependency? something that implements the full spec?

**Comment by benswinburne** [comment]
I'd hazard a guess that https://www.npmjs.com/package/conventional-commits-parser would be a good bet.

That said, there is currently an open issue https://github.com/conventional-changelog/conventional-changelog/issues/648 describing similar ! behaviour to above. However, with this library even in it's present state the issue can be resolved by passing in a different matcher which includes ! as a parser option described in [this comment](https://github.com/conventional-changelog/conventional-changelog/issues/648#issuecomment-704867077)

The resulting structured data from the parser should easily be sufficient for use in Auto I believe?

```
{ type: 'feat',
  scope: 'scope',
  subject: 'broadcast $destroy event on scope destruction',
  merge: null,
  header: 'feat(scope): broadcast $destroy event on scope destruction',
  body: null,
  footer: 'Closes #1',
  notes: [],
  references:
   [ { action: 'Closes',
       owner: null,
       repository: null,
       issue: '1',
       raw: '#1',
       prefix: '#' } ],
  mentions: [],
  revert: null }
{ type: 'feat',
  scope: 'ng-list',
  subject: 'Allow custom separator',
  merge: null,
  header: 'feat(ng-list): Allow custom separator',
  body: 'bla bla bla',
  footer: 'BREAKING CHANGE: some breaking change.\nThanks @stevemao',
  notes:
   [ { title: 'BREAKING CHANGE',
       text: 'some breaking change.\nThanks @stevemao' } ],
  references: [],
  mentions: [ 'stevemao' ],
  revert: null }
```



[@commitlint/parse](https://www.npmjs.com/package/@commitlint/parse) appears to be a [wrapper](https://github.com/conventional-changelog/commitlint/blob/master/%40commitlint/parse/src/index.ts) around conventional-commits-parser with types.

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v9.60.0`](https://github.com/intuit/auto/releases/tag/v9.60.0) :rocket:

**Comment by hipstersmoothie** [comment]
No prob! def needed an update and now that we're on a package the much of the ecosystem uses it unlocks using conventional-changelog presets in the plugin:

https://intuit.github.io/auto/docs/generated/conventional-commits#options
